### DISCUSSION TITLE
Titles and Markdown

### DISCUSSION THREAD

**Discussion by bobgeorgethe3rd** [freesewing/freesewing#1773] [Support] [answered]
Hello so I have a question about title placements and title naming.
I was under the impression that we were not suppose to add the title directly to the pages as it is kept in another file but has that had changed recently and they are now required as I saw they were kept on the recent Lunetius update?

I know we keep umming and erring about updating the titles so that they include the Patterns Name like how Holmes docs do. I am full prepared to go through and update all of the titles in one swing if it is something we would like, I just need to know were the titles are kept and I'll do it like now.

**Comment by joostdecock** [comment, marked as answer]
Currently, titles for some pages (typically the sub-pages of pattern documentation, option documentation and so on) are injected by the frontend code. As such, setting them in the markdown was traditionally discouraged.
(The titles are sourced from the translation files)

However, we've started working on new frontend code, and this entails some changes. One of them is the support for better open graph images like this for example: 

![](https://canary.backend.freesewing.org/og-img/en/dev/guides/markdown)

These images are auto-generated based on the markdown content. So, if there's no title in the markdown, we can't generate them.

There's also the fact that all this abstraction (titles come from there except for those pages where they come from there) adds a lot of complexity making it harder for people to wrap their head around how things work together.

So, since we've already deployed freesewing.dev using the new system, the rule is now that:

 - Titles should be set in frontmatter
 - Each page should start with (at least one) paragraph that is a good summary (as it will be used in the OG image).

That being said, the titles for many documentation pages (patterns, options, and measurements) can still be determined from their translations. So rather than go through and update them all, we should write some script that does this in an automated way. That will also help us keep titles in sync.
Let's not forget, the main reason we use titles from the translation is that we want to ensure consistency of options/measurements between the various places they are used (in menus, their description/title from from translations files, but in docs, it comes from markdown). So we need an automated way to ensure this consistency, or it will become a nightmare to maintain.

I'm occupied with the new development environment at the moment, and I'm not sure how easy it would be for others to automate this.
If you'd like to take a stab at it, go ahead. If not, please allow me to finish one thing before starting the next because context switching is expensive.

**Reply by bobgeorgethe3rd** [reply]
This makes sense. There's no rush, since I work a lot in the markdown I was like oh does need to be done? And the answer is yes but not right now and not manually :) I may look into automating but I do not have any experience with this sort of thing but hey maybe it's a learning oppurtunity :) Thank You :)

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
